### PR TITLE
SQL: have LIKE/RLIKE use wildcard and regexp queries

### LINKED ISSUE #40557: SQL: LIKE not treating whitespaces correctly -- version 6.7.0
**Elasticsearch version** (`bin/elasticsearch --version`): `6.7.0`

**Plugins installed**: []

**JVM version** (`java -version`): 
```
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**OS version** (`uname -a` if on a Unix-like system): `MacOS Mojave 10.14.4`

**Description of the problem including expected versus actual behavior**:
I noticed that when using `LIKE`, in this case `LIKE 'B 4E%'`, under the hood it's no longer using a wildcard query, but instead `query_string` (from the translate API):
```
"bool" : {
      "must" : [
        {
          "query_string" : {
            "query" : "B 4E*",
            "fields" : [
              "station.keyword^1.0"
            ],
            "type" : "best_fields",
            "default_operator" : "or",
            "max_determinized_states" : 10000,
            "enable_position_increments" : true,
            "fuzziness" : "AUTO",
            "fuzzy_prefix_length" : 0,
            "fuzzy_max_expansions" : 50,
            "phrase_slop" : 0,
            "escape" : false,
            "auto_generate_synonyms_phrase_query" : true,
            "fuzzy_transpositions" : true,
            "boost" : 1.0
          }
        },
```
Problem I'm having is that my `LIKE` clause above has whitespaces and it's not treated correctly as it was before with the `wildcard`. I believe that because of the whitespace I have above, the query is trying to match on `B` or `4E%`, but I want `B 4E%`...

The hacky solution I came up with was to replace the whitespaces with underscore, so `B 4E%` became `B_4E%` which then gave me the expected response data. 

Can anyone please advise on this issue? Thanks

### PR DESCRIPTION
**Pull request by astefan**
LIKE and RLIKE should only create and use `wildcard` (for LIKE) and `regexp` (for RLIKE) queries. Some time ago, we allowed inexact field for these operators and created `query_string` queries in those case. SQL should work on exact fields for LIKE and RLIKE (which does already), but the queries generated were incorrect. This PR fixes this bug.

Fixes https://github.com/elastic/elasticsearch/issues/40557.

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java [hunk 1]
```diff
@@ -470,42 +471,30 @@ private static String topAggsField(AggregateFunction af, Expression e) {
...
-            String target = null;
+            String targetFieldName = null;
 
             if (e.field() instanceof FieldAttribute) {
-                target = nameOf(((FieldAttribute) e.field()).exactAttribute());
+                targetFieldName = nameOf(((FieldAttribute) e.field()).exactAttribute());
             } else {
-                throw new SqlIllegalArgumentException("Scalar function ({}) not allowed (yet) as arguments for LIKE",
-                        Expressions.name(e.field()));
+                throw new SqlIllegalArgumentException("Scalar function ({}) not allowed (yet) as arguments for "
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java]: We use `[]` inside our error messages: "Scalar Function [{}] not allowed..."

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java [hunk 2]
```diff
@@ -470,42 +471,30 @@ private static String topAggsField(AggregateFunction af, Expression e) {
...
+                        + e.getClass().getSimpleName().toUpperCase(Locale.ROOT), Expressions.name(e.field()));
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java]: We need to get the function name from the source() if I'm not mistaken.

**Review by matriv — APPROVED** [review]
LGTM left a comment

**Pull request merged**